var/spack/repos/builtin/packages/gperftools/package.py

```diff
@@ -26,7 +28,8 @@ class Gperftools(AutotoolsPackage):
     variant('sized_delete', default=False, description="Build sized delete operator")
     variant('dynamic_sized_delete_support', default=False, description="Try to build run-time switch for sized delete operator")
     variant('debugalloc', default=True, description="Build versions of libs with debugalloc")
-    variant('libunwind', default=True, description="Enable libunwind linking")
+    # libunwind is not supported on Darwin
+    variant('libunwind', default=sys.platform != 'darwin', description="Enable libunwind linking")
```

Comment:
Does `not supported` actually mean that it is not possible to have it enabled on Darwin? In case, I would add also a `conflict`.

Reply:
Would:
```suggestion
    variant('libunwind', default=True, when='platform=linux', description="Enable libunwind linking")
```
work too, without adding the conflict?

Reply:
Yes, `libunwind` declares a conflict on Darwin. It's not possible to install it there. Do we need to repeat this in all packages that use `libunwind`?

(I notice that the concretizer didn't notice this conflict; I think it should have.)

Reply:
A added the conflict.

Reply:
@alalazo When you say `platform=linux`, should that be `platform!=darwin` instead? What syntax is supported here?

Reply:
The suggestion from @alalazo looks correct, you can avoid the `conflict` by saying "when the variant is enabled". AFAIK `when` directive accepts "spack spec language", so there should not be any logical operator in there and it should be for this reason why @alalazo is suggesting to specify `platform=linux`.

Reply:
@alalazo ping

Reply:
Done.

Reply:
I'm fine with the change as is if it works for @eschnett.

I think the question was though if `platform!=darwin` is equivalent to `platform=linux` (independent of if it's a `conflict` or `when`)? Based on https://spack.readthedocs.io/en/latest/spack.platforms.html it seems like it's not because there's also `windows` and then the `libunwind` variant would be available for `windows` as well. But then I don't know if gperftools can even be used on windows, so it might be irrelevant.